dataclients/kubernetes/definitions/routegroupvalidator.go

```diff
@@ -0,0 +1,339 @@
+package definitions
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"errors"
+
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/loadbalancer"
+	"gopkg.in/yaml.v2"
+)
+
+type RouteGroupValidator struct {
+	FilterRegistry filters.Registry
+}
+
+var rgv = &RouteGroupValidator{}
+
+func (rgv *RouteGroupValidator) Validate(item *RouteGroupItem) error {
+	err := rgv.BasicValidation(item)
+	if rgv.FilterRegistry != nil {
+		err = errors.Join(err, rgv.FiltersValidation(item))
+	}
+	return err
+}
+
+func (rgv *RouteGroupValidator) BasicValidation(item *RouteGroupItem) error {
+	return item.validate()
+}
+
+func (rgv *RouteGroupValidator) FiltersValidation(item *RouteGroupItem) error {
+	return validateRouteGroupFilters(item, rgv.FilterRegistry)
+}
+
+func (rgv *RouteGroupValidator) ValidateList(rl *RouteGroupList) error {
```

Comment:
Do we need ValidateList? If not - lets not have it, we can always add when needed

Reply:
I see we use it in ValidateRouteGroups